src/libtexture/texturesys.cpp

```diff
@@ -82,6 +82,12 @@ static const OIIO_SIMD4_ALIGN vbool4 channel_masks[5] = {
 TextureSystem*
 TextureSystem::create(bool shared, ImageCache* imagecache)
 {
+// Because the shared_texturesys is never deleted (by design)
+// we silence the otherwise useful compiler warning on newer GCC / Clang versions
+#pragma GCC diagnostic push
+#if __GNUC__ > 10
+#    pragma GCC diagnostic ignored "-Wmismatched-new-delete"
+#endif
```

Comment:
We have macros defined in platform.h for these things. In the parlance we use through the rest of the code base, this would be

```c++
OIIO_PRAGMA_WARNING_PUSH
#if OIIO_GNUC_VERSION > 100000
    OIIO_GCC_ONLY_PRAGMA(diagnostic ignored "-Wmismatched-new-delete")
#endif
...
OIIO_PRAGMA_WARNING_POP
```

I admit that for this small example, this is not especially any more clear than the raw pragmas you used. But the point is that it'll always work properly, no future non-gcc compiler will complain about those GCC pragma (which I am kinda superstitious about, even though CI shows it appears to be fine on the compilers we test).

Reply:
Awesome, I've updated the pull request to use the existing macros.
I also edited the comment as this now only applies to GCC (as your macros explicitly single out that compiler). Usually clang respects gcc pragmas, but I don't have a clang toolchain easily available, and I guess the existing CI builders haven't complained so far